compiler/codegen/codegen.cpp

```diff
@@ -2712,7 +2712,10 @@ void makeBinary(void) {
     const char* command = astr(astr(CHPL_MAKE, " "),
                                makeflags,
                                getIntermediateDirName(), "/Makefile");
-    mysystem(command, "compiling generated source");
+    int status = mysystem(command, "compiling generated source");
+    if (status != 0) {
+      INT_FATAL("Error compiling generated sources");
+    }
```

Comment:
I'm confused about why `ignoreStatus=false` being the default argument didn't cause it to halt when the exit code was not zero. It looks to me like the mysystem implementation is generally intended to halt inside of it if the exit code is nonzero.

Reply:
(Which is to say, this fixes the bug in one place, but maybe we need to fix it inside of mysystem to cover other places).

Reply:
Oh, interesting Michael.  I'd forgotten about this and only saw that mysystem() returned `int`s in one of its instances, and that that fixed the issue for me, so added this without looking deeper into its implementation.  I'll take another look.

Reply:
Aha, it looks like https://github.com/chapel-lang/chapel/pull/18594 broke that behavior.  In checking back to 1.10, I was only checking in the sources that the return code of mysystem() was ignored... I'd forgotten that it handled errors internally (and it's just annoying enough to test manually that I didn't bother).